merge/object.go

```diff
@@ -0,0 +1,43 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package merge
```

Comment:
Seems like this package might be turning into the top-level interface for this code. That's OK; I'm not sure the name is right though.

Reply:
Yeah, I have the feeling that maybe this package will end-up being the only visible package from the outside, and everything else could be turned into an internal package. And I agree that the name sucks. I do intent to work on moving things around before we start merging in kubernetes to make sure that we don't over-expose implementation details.